utils/global-state-update-gen/src/generic/testing.rs

```diff
@@ -570,6 +570,104 @@ fn should_replace_one_validator() {
     assert_eq!(update.len(), 10);
 }
 
+#[test]
+fn should_replace_one_validator_with_unbonding() {
+    let mut rng = TestRng::new();
+
+    let validator1 = PublicKey::random(&mut rng);
+    let validator2 = PublicKey::random(&mut rng);
+
+    let mut reader = MockStateReader::new().with_validators(
+        vec![(
+            validator1.clone(),
+            U512::from(101),
+            ValidatorConfig {
+                bonded_amount: U512::from(101),
+                ..Default::default()
+            },
+        )],
+        &mut rng,
+    );
+
+    // we'll be updating the validators set to only contain validator2
+    let config = Config {
+        accounts: vec![AccountConfig {
+            public_key: validator2.clone(),
+            balance: Some(U512::from(102)),
+            validator: Some(ValidatorConfig {
+                bonded_amount: U512::from(102),
+                delegation_rate: None,
+                delegators: None,
```

Comment:
Maybe also use `default()` here, or `None` above for validator 1, just be be consistent.